src/vos/evtree.c

```diff
@@ -2433,6 +2443,16 @@ evt_ssof_insert(struct evt_context *tcx, struct evt_node *nd,
 			return -DER_NOMEM;
 		ne->ne_child = desc_mmid.oid.off;
 		desc = evt_tmmid2ptr(tcx, desc_mmid);
+		rc = vos_dtx_register_record(evt_umm(tcx),
```

Comment:
same here, this function is always called?

Reply:
It is always called when create new evt_desc, but if we disable DTX, then the vos_dtx_register_record() will bypass most of the DTX logic and only mark the evt_desc::dc_dtx as NULL, then return directly.